gh_pages/_source/session8/Motion-Planning-with-Tesseract.rst

```diff
@@ -0,0 +1,435 @@
+
+Implement the Descartes Planner Profile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Create the planner profile:
+	
+   Within ``snp_motion_planning/src/planner_profiles.hpp``, find the section
+
+   .. code-block:: c++
+
+      /* =======================
+       * Fill Code: DESCARTES 
+       * =======================*/
+
+   We must fist set up some configurations we want our Descartes planner to follow. The following block specifies the number of threads we need for the planner, if we allow redundant joint solutions, and whether or not to allow collisions.
+   Copy and paste the following:
+
+   .. code-block:: c++
+
+   	auto profile = std::make_shared<tesseract_planning::DescartesDefaultPlanProfile<FloatType>>();
+  		profile->num_threads = static_cast<int>(std::thread::hardware_concurrency());
+	  	profile->use_redundant_joint_solutions = false;
+	  	profile->allow_collision = false;
+	  	profile->enable_collision = true;
+	  	profile->enable_edge_collision = false;
+
+   Now we will also specify our state and edge evaluators. The state evaluator looks at a given state and gives the state both a cost and a pass or fail. A few things we may choose to evaluate are whether or not the state is valid, the cost of the state, and any biases we may want to give it. The edge evaluator works similarly but evaluates a transition between states. 
+
+   Copy and past the following below the previous block:
+
+   .. code-block:: c++
+
+	  	// Use the default state and edge evaluators
+  		profile->state_evaluator = nullptr;
+  		profile->edge_evaluator = [](const tesseract_planning::DescartesProblem<FloatType>& prob) ->
+      		typename descartes_light::EdgeEvaluator<FloatType>::Ptr {
+        		auto eval = std::make_shared<descartes_light::CompoundEdgeEvaluator<FloatType>>();
+
+        		// Nominal Euclidean distance
+        		eval->evaluators.push_back(std::make_shared<descartes_light::EuclideanDistanceEdgeEvaluator<FloatType>>());
+
+        		return eval;
+      		};
+      	profile->vertex_evaluator = nullptr;
+
+   Finally, we set the ``target_pose_sampler`` which takes a given function for sampling. In our example, we specify our pose sampling to allow any rotation along the z-axis as it will not impact our final results. Note that Descartes can only work in discrete space so we are only sampling at increments of 10 degrees around the z-axis.
+
+   Copy and past the following below the previous block:
+
+   .. code-block:: c++
+  		
+  		profile->target_pose_sampler =
+      		std::bind(tesseract_planning::sampleToolZAxis, std::placeholders::_1, 10.0 * M_PI / 180.0);
+
+  		return profile;
+
+#. Add the planner to the planning server:
+   
+   Within ``snp_motion_planning/src/planner_profiles.hpp``, find the section
+
+   .. code-block:: c++
+
+      /* ========================================
+       * Fill Code: ADD CUSTOM PLANNER PROFILES
+       * ========================================*/
```

Comment:
This line isn't in planner_profiles.hpp . Did you mean: 
  /* ==============================
   * Fill Code: TRAJOPT COMPOSITE 
   * ==============================*/
     /* ==========================
   * Fill Code: TRAJOPT PLAN 
   * ==========================*/
   or 
     /* =======================
   * Fill Code: OMPL 
   * =======================*/
   ?